pandas/tests/plotting/test_groupby.py

```diff
@@ -10,11 +10,52 @@
 import numpy as np
 
 from pandas.tests.plotting.common import TestPlotBase
+import pytest
 
 
 @td.skip_if_no_mpl
 class TestDataFrameGroupByPlots(TestPlotBase):
 
+    @pytest.mark.parametrize('equal_bins, bins, expected1, expected2', (
+        (True, 10,
+         [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4],
+         [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4]),
+        (True, None,
+         [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4],
+         [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4]),
+        (True, [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4, 3.],
+         [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4],
+         [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4]),
+        (False, [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4, 3.],
+         [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4],
+         [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4]),
+        (False, 10,
+         [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4],
+         [-1., -0.8, -0.6, -0.4, -0.2, 0., 0.2, 0.4, 0.6, 0.8]),
+        (False, None,
+         [-3., -2.4, -1.8, -1.2, -0.6, 0., 0.6, 1.2, 1.8, 2.4],
+         [-1., -0.8, -0.6, -0.4, -0.2, 0., 0.2, 0.4, 0.6, 0.8])
+    ))
+    def test_hist_bins_match(self, equal_bins, bins, expected1, expected2):
+        # GH-22222
+        df = DataFrame(np.append(np.linspace(-3, 3, 100),
+                                 np.linspace(-1, 1, 100)),
+                       columns=['value'])
+        df['group'] = [0] * 100 + [1] * 100
+        g = df.groupby('group')['value']
+        ax = g.hist(bins=bins, alpha=0.7, equal_bins=equal_bins)[0]
+
+        # x-axis (leftmost) hist bar coordinates:
+        # group0: points[:num_bins]
+        # group1: points[num_bins:]
+        num_bins = len(expected1)
+        points = np.array([patch.get_bbox().get_points()
+                           for patch in ax.patches])[:, 0, 0]
+
+        tm.assert_almost_equal(points[:num_bins], np.array(expected1))
```

Comment:
Would prefer to use `tm.assert_numpy_array_equal` instead here and on line below